### PR TITLE
Adds Grid Velocity Term in the Euler/Symmetry BC

### LINKED ISSUE #828: Different results on the same testcase of unsteady ONERA (v6.0.1 vs v6.2.0)
**Description of the issue**
Hello everybody,

I'm currently running the same testcase (Unsteady Euler ONERAM6 AoA=3deg) on two different versions of SU2 ( version 6.0.1 and 6.2.0). What I noticed is a different evolution in time of coefficients, Cl and Cd as given in the figure. Both the codes look at convergence and the input files (despite the version of the code being  different) are the same. I attach the two screen outputs for comparison. I'm running SU2_CFD through the **parallel_computation.py** with only one core (using different number of cores doesn't affect the result anyway).

Screenshot 2019-11-28 at 17 53 42

[Summary_singlecore_6_2_0.txt](https://github.com/su2code/SU2/files/3902407/Summary_singlecore_6_2_0.txt)
[Summary_singlecore_6_0_1.txt](https://github.com/su2code/SU2/files/3906057/Summary_singlecore_6_0_1.txt)

**To Reproduce the problem**
To reproduce the problem I attach the config files. The mesh is available in the repository: **mesh_ONERAM6_inv.su2**

Note: both versions are/were on the develop branches. Latest one is up to date.

[config_CFD_6_0_1.txt](https://github.com/su2code/SU2/files/3906061/config_CFD_6_0_1.txt)
[config_CFD_6_2_0.txt](https://github.com/su2code/SU2/files/3902415/config_CFD_6_2_0.txt)

Im running the two versions on different machines with same specs:
**Details for the new version**
 - OS: [Ubuntu 16.04.6 LTS]
 - C++ compiler and version: [g++ 5.4.0]
 - MPI implementation and version: [Open MPI 1.10.2]
 - SU2 Version: [v6.2.0]

**Details for the old version**
 - OS: [Ubuntu 16.04.6 LTS]
 - C++ compiler and version: [g++ 5.4.0]
 - MPI implementation and version: [Open MPI 1.10.2]
 - SU2 Version: [v6.0.1]

Thank you and best,
Rocco

### LINKED ISSUE #997: Caradonna_tung test case is not initialazing solution
Hello. I have been trying to work with test case "Caradonna_tung". I have obtained the latest config file and mesh file i believe. Hovewer, when i run the solver, it gives value "zero" for everything and seemingly converged after only 10 iterations. I believe i could not initialize the solver. Can anyone help me with it ? Thank you.

[caradonna_tung.zip](https://github.com/su2code/SU2/files/4664054/caradonna_tung.zip)

### PR DESCRIPTION
**Pull request by economon**
## Proposed Changes
*Give a brief overview of your contribution here in a few sentences.*
 
Adjustment to include grid velocity in the reflected BC state for dynamic grid calculations. 

Checking to see how this behaves with the regressions - there are some other tests ongoing too.

## Related Work
*Resolve any issues (bug fix or feature request), note any related PRs, or mention interactions with the work of others, if any.*

Resolves #828 
Resolves #997 

## PR Checklist
*Put an X by all that apply. You can fill this out after submitting the PR. If you have any questions, don't hesitate to ask! We want to help. These are a guide for you to know what the reviewers will be looking for in your contribution.*

- [x] I am submitting my contribution to the develop branch.
- [x] My contribution generates no new compiler warnings (try with the '-Wall -Wextra -Wno-unused-parameter -Wno-empty-body' compiler flags).
- [x] My contribution is commented and consistent with SU2 style.
- [ ] I have added a test case that demonstrates my contribution, if necessary.
- [ ] I have updated appropriate documentation (Tutorials, Docs Page, config_template.cpp) , if necessary.

### CODE REVIEW

File: SU2_CFD/src/solvers/CEulerSolver.cpp [hunk 1]
```diff
@@ -6874,6 +6874,12 @@ void CEulerSolver::BC_Sym_Plane(CGeometry      *geometry,
             velocity in normal direction: v_r = v - 2 (v*n)n ---*/
       ProjVelocity_i = nodes->GetProjVel(iPoint,UnitNormal);
 
+      /*--- Adjustment to v.n due to grid movement. ---*/
+      if (dynamic_grid) {
+          for (iDim = 0; iDim < nDim; iDim++)
+            ProjVelocity_i -= geometry->nodes->GetGridVel(iPoint)[iDim]*UnitNormal[iDim];
+      }
```
Comment (pcarruscag) [inline SU2_CFD/src/solvers/CEulerSolver.cpp:6881]: I added a geometry toolbox that defines dot products and other functions, I think it makes these bits of code more expressive.

**Comment by talbring** [comment]
Whats the status here? Release v7.0.5 will be out tmrw :)

**Comment by pcarruscag** [comment]
Actually, shouldn't we reflect the grid velocity of the fictitious jPoint? instead of modifying the reflection of the flow velocity terms?
I thought the flow velocity projection we do here was purely geometric, i.e. it does not have the usual physical meaning.

**Comment by economon** [comment]
At a flow tangency wall when we have grid movement, we need to satisfy (v - v_grid).n = 0 instead of the usual v.n = 0. So the modification should give us the correct quantity that we need to reflect (so that it is zero at the wall). Curious readers can flip to pg 65 :).

File: SU2_CFD/src/solvers/CEulerSolver.cpp (hunk 1 above)
Comment (economon) [reply SU2_CFD/src/solvers/CEulerSolver.cpp]: Nice, we should use those as much as possible. Will help minimize bugs.

My only complaint is the long namespace name.. maybe we could standardize on some acronyms like AD, GT, PT, etc. Just a thought

Comment (pcarruscag) [reply SU2_CFD/src/solvers/CEulerSolver.cpp]: Had to look it up, it is possible to alias namespaces (https://en.cppreference.com/w/cpp/language/namespace_alias).

**Comment by economon** [comment]
Let's get this merged in, but leave the issues open for final comments about these changes.

**Pull request merged**